src/Microsoft.Diagnostics.Monitoring.RestServer/Controllers/DiagController.cs

```diff
@@ -28,14 +28,16 @@ namespace Microsoft.Diagnostics.Monitoring.RestServer.Controllers
     [ApiController]
     [HostRestriction]
     [Authorize(Policy = AuthConstants.PolicyName)]
+    [ProducesResponseType(typeof(ValidationProblemDetails), StatusCodes.Status400BadRequest)]
+    [ProducesResponseType(StatusCodes.Status401Unauthorized)]
```

Comment:
Just for my own understanding, the OpenApi generator uses these attributes to figure out where to apply things like the UnauthorizedResponseFilter correct?

Reply:
These attributes tell the API Explorer what types of status codes (other than 200) the routes will return. Then the OpenAPI generator will query the API Explorer via Swashbuckle to get this information, which will write out these as responses for each route. However, the content type of the responses are dictated by the ProducesAttribute (and if that does not exist, the default content types).

Using the logs route as an example, it would say that 401 produces application/x-ndjson, text/event-stream, and application/problem_json. These are not correct and there is no way to describe using attribution that certain responses only have certain content types. So I use IOperationFilter, such as UnauthorizedResponseFilter to fix up the response (specifically, clear it out and set a reference to a global response description).